avocado/plugins/assets.py

```diff
@@ -0,0 +1,262 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2019
+# Authors: Willian Rampazzo <willianr@redhat.com>
+
+"""
+Assets subcommand
+"""
+
+import ast
+import os
+
+from avocado.core import data_dir
+from avocado.core import exit_codes
+from avocado.core import safeloader
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import CLICmd
+from avocado.utils.asset import Asset
+from avocado.utils import data_structures
+
+
+class FetchAssetHandler(ast.NodeVisitor):  # pylint: disable=R0902
```

Comment:
Do you use any customized pylint config file? In my system, I have a warning when I remove the disable statement: `assets.py|31 col 1 warning| [too-many-instance-attributes] Too many instance attributes (8/7) [python/pylint]`. Here just 7 are allowed.

Reply:
No, I don't... but it may be related to a version mismatch.  Are you running pylint 2.3.0 (as defined by `requirements-selftests.txt`) or a later version?

Reply:
I'm running 2.3.0:
```
$ pylint --version
pylint 2.3.0
astroid 2.2.0
Python 3.7.4 (default, Jul  9 2019, 16:32:37) 
[GCC 9.1.1 20190503 (Red Hat 9.1.1-1)]
```

Reply:
Same warning when I run pylint manually instead of using it at the editor:
```
$ pylint avocado.dev/avocado/plugins/assets.py 
************* Module avocado.plugins.assets
avocado.dev/avocado/plugins/assets.py:31:0: R0902: Too many instance attributes (8/7) (too-many-instance-attributes)

------------------------------------------------------------------
Your code has been rated at 9.92/10 (previous run: 9.92/10, +0.00)
```

Reply:
OK, when running pylint manually I see it (I will usually run ./selftests/inspekt-lint.sh which wraps it).  I'm not opposed to ignoring those checks, but I noticed that if you turn `self.pattern` into a class attribute, you'd be under the 7 instance attributes limit (and would still not need to ignore the check).

Reply:
That makes sense :)